core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/transfer/TransferProcessManagerImpl.java

```diff
@@ -523,29 +515,38 @@ private void processProviderRequest(TransferProcess process, DataRequest dataReq
                 observable.invokeForEach(l -> l.error(process));
             }
         }
+
+        return true;
     }
 
-    private void sendConsumerRequest(TransferProcess process, DataRequest dataRequest) {
+    private boolean sendConsumerRequest(TransferProcess process, DataRequest dataRequest) {
         process.transitionRequested();
         transferProcessStore.update(process);   // update before sending to accommodate synchronous transports; reliability will be managed by retry and idempotency
         observable.invokeForEach(l -> l.requested(process));
         dispatcherRegistry.send(Object.class, dataRequest, process::getId)
                 .thenApply(o -> {
-                    transitionRequestAck(process.getId());
-                    transferProcessStore.update(process);
+                    // TODO: what's the point of this state transition?
+                    transitionToRequestAck(process.getId());
                     return o;
                 })
                 .whenComplete((o, throwable) -> {
                     if (o != null) {
                         monitor.info("Object received: " + o);
-                        if (dataRequest.getTransferType().isFinite()) {
-                            process.transitionInProgress();
-                        } else {
-                            process.transitionStreaming();
-                        }
-                        transferProcessStore.update(process);
+                        transitionToInProgress(transferProcessStore.find(process.getId()));
```

Comment:
Any particular reason why you're performing the `transferProcessStore.find(...)` directly here as opposed to: inside the `transitionToInProgress` method?

Reply:
Yeah, I'm not that happy with this solution, I did like this to have the opportunity to use the `transitionToInProgress` also from `processProviderRequest` and `processAckRequested`. Added a null check though.

Reply:
I think the if condition could be moved in a method inside `TransferProcess`. Doing this the method `transitionToInProgress` would be superfluous.

Reply:
@paullatzelsperger I added a method on `TransferProcess` that handles the choice between "in progress" and "streaming". What do you think?